src/librustc_lint/lib.rs

```diff
@@ -318,6 +318,7 @@ fn register_builtins(store: &mut LintStore, no_interleave_lints: bool) {
     store.register_renamed("async_idents", "keyword_idents");
     store.register_renamed("exceeding_bitshifts", "arithmetic_overflow");
     store.register_renamed("redundant_semicolon", "redundant_semicolons");
+    store.register_renamed("intra_doc_link_resolution_failure", "intra_doc_link_resolution_failures");
```

Comment:
Since we're renaming it anyway - `intra_doc_link_resolution_failures` seems really long to me. Can we make it shorter? I've been typing `intra_doc_resolution_failures` by accident in the tests, that seems just as descriptive to me.

Reply:
Another alternative is `broken_intra_doc_links`.

Reply:
Works for me!

Reply:
Was just about to leave a comment like this! `broken_intra_doc_links` sounds good to me. "Resolution" always bothered me as too jargony.

Reply:
@Manishearth do you mind changing this to `broken_intra_doc_links`? I can make a PR if you're tired of messing with this one.

Reply:
I like this too, done.